source/nordic/nrf52833/FlashDev.c

```diff
@@ -0,0 +1,38 @@
+/* Flash OS Routines
+ * Copyright (c) 2009-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FlashOS.H"
+
+#define FLASH_DRV_VERS (0x0100+VERS)   // Driver Version, do not modify!
+#define DEVICE_NAME    "nRF52833"
+
+struct FlashDevice const FlashDevice  =  {
+    FLASH_DRV_VERS,             // Driver Version, do not modify!
+    DEVICE_NAME,                // Device Name
+    ONCHIP,                     // Device Type
+    0x00000000,                 // Flash start address
+    0x00080000,                 // Flash total size (512 KB)
+    0x1000,                     // Programming Page Size (4 KB)
```

Comment:
I wasn't sure if this should be the flash page size (4KB) or the minimum size data block that can be written into flash (a 4 bytes word).

The nRF51 version has configured this value to `4` instead.

And looking at the Python template this values ends up going to both `page_size` and `min_program_length`, so I'm not 100% sure.
https://github.com/pyocd/FlashAlgo/blob/36e24a43cfdc12a26d4c9057ad33a15b83556f39/scripts/templates/py_blob_orig.tmpl#L37-L41

Reply:
Ups, sorry I though I submitted this message as soon as the PR as up, but it was stuck in "pending review".
@flit I assume this is correct, but it'd be great if you could confirm

Reply:
It could go either way, honestly.

This is a poorly defined part of the flash algo spec. The Keil team use it pretty much as a preferred software-defined programming unit, usually quite a bit larger than the hardware write size for performance reasons. But most of their flash algo implementations will allow you to program any size data as long as it's aligned to the hardware-defined write size. There is no way to set the hardware write size in a flash algo definition.

So, you may actually want to change it. PyOCD and DAPLink (?) use that value as the minimum write size, by default, since there is no way to declare what the hardware will allow.